### DISCUSSION TITLE
Trouble creating/importing site template

### DISCUSSION THREAD

**Discussion by OwenParsons1** [pnp/powershell#2507] [Q&A]
Currently trying to provision a site using a template1.xml file which is generated from a template site using the command:
**_Get-PnPSiteTemplate -Force -Out template1.xml_**

From here we are then creating our new site using:
**_New-PnPSite -Type TeamSiteWithoutMicrosoft365Group -Title $Title -Url $SiteURL_**
(The type is the same as our template site)

The script will then connect to the site:
**_Connect-PnPOnline -URL $SiteURL -Interactive_**

After connecting, the script will begin to invoke the previously generated template1.xml file onto our new site.
**_Invoke-PnPSiteTemplate -Path template1.xml_**

This is where we are running into issues. The script then throws up the following error:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/116895038/198622883-5cff6c19-53d2-428b-b0b8-3704b8cc93d2.png">

We have taken a look into the generated template1.xml file to see if the file contains the column which the script says does not exist and we are able to find it here:

**_Field Type="Choice" DisplayName="TK Style" Required="FALSE" EnforceUniqueValues="FALSE" Indexed="FALSE" Format="Dropdown" FillInChoice="FALSE" Group="TK Columns" ID="{70d06a52-31c2-4ba7-b360-b4e18b579273}" SourceID="{3c25c9b6-2bc1-461a-83fb-d84101990711}" StaticName="TK_x0020_Style" Name="TK_x0020_Style"
					CHOICES
						CHOICE>Style A</CHOICE
						CHOICE>Style B</CHOICE
						CHOICE>Style C</CHOICE_**

We have also looked on the site we are using as a template to make sure that the column definitely exists, and we are able to find it where we expected listed under a content type:
![image](https://user-images.githubusercontent.com/116895038/198626109-b76dd16b-82ac-4a98-af97-5cf1fc3218c8.png)

We then looked on our generated site and also found it created there under the same content type as our template site.
![image](https://user-images.githubusercontent.com/116895038/198626656-6bee540f-a51e-4c7e-94b8-5bb00fa16349.png)

I would appreciate any guidance on resolving this issue.

**Comment by OwenParsons1** [comment]
Bump

**Comment by AndersRask** [comment]
Try and enable trace log in the session you are running the script in, and see if it gives any hints.

`Set-PnPTraceLog -On -Level Information -LogFile tracelog.txt`

You can also try and seperate columns and content types in seperate invokes with seperate handlers to ensure this isnt race conditions (another way to test that is to re-run the template, and see if it finishes, as the cmdlets are idempotent)

**Reply by OwenParsons1** [reply]
Thanks for replying.

I've enabled the trace log and I can now see where the issue is

**_pwsh Information: 0 : 2022-11-11 10:23:37.0485	[PnP.Framework]	[0]	[Information]	Creating content type 'Catalog-Item Reuse' (0x010100C568DB52D9D0A14D9B2FDCC96666E9F2007948130EC3DB064584E219954237AF3900B46186789C3140CC85BE610336E86BBB).	0ms	
pwsh Error: 0 : 2022-11-11 10:23:37.1421	[PnP.Framework]	[0]	[Error]	ExecuteQuery threw following exception: Microsoft.SharePoint.Client.ServerException: The parent content type specified by content type identifier 0x010100C568DB52D9D0A14D9B2FDCC96666E9F2007948130EC3DB064584E219954237AF39 does not exist.
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream)
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponse()
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryToServerAsync(ChunkStringBuilder sb)
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientRuntimeContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
ServerErrorCode: -2146232832
ServerErrorTypeName: Microsoft.SharePoint.SPException_**

We keep getting the same error on this content type called 'Catalog-Item Reuse'
Is there a way of excluding a specific content type on an invoke because this one seems to be stopping everything else from being created.

**Reply by AndersRask** [reply]
It looks like you are missing the parent content type of Catalog-Item

you can see the inheritance as part of the Content Type id below
If you include the parent CT in your template xml it should work.

If you want to remove it, you need to also remove any references in your list templates too.

```
0x010100C568DB52D9D0A14D9B2FDCC96666E9F2007948130EC3DB064584E219954237AF39 
0x010100C568DB52D9D0A14D9B2FDCC96666E9F2007948130EC3DB064584E219954237AF3900B46186789C3140CC85BE610336E86BBB
                                                                          ^^
```

**Comment by BaronSparky** [comment]
Is there a reason why you are not using tenant templates? You wouldn't need to necessarily create the site first; you could do it all from the invoke.

**Reply by OwenParsons1** [reply]
We've had a customer create a site and be able to copy it using a script, so this was the method we used.
